waspc/data/Generator/templates/react-app/src/storage.ts

```diff
@@ -0,0 +1,44 @@
+export type DataStore = {
+    set(key: string, value: unknown): void;
+    get(key: string): unknown | undefined;
+    remove(key: string): void;
+    clear(): void;
+};
+
+function createLocalStorageDataStore(prefix: string): DataStore {
+    return {
+        set(key, value) {
+            ensureLocalStorageIsAvailable();
+            localStorage.setItem(`${prefix}:${key}`, JSON.stringify(value));
+        },
+        get(key) {
+            ensureLocalStorageIsAvailable();
+            const value = localStorage.getItem(`${prefix}:${key}`);
+            try {
+                return value ? JSON.parse(value) : undefined;
+            } catch (e: any) {
+                return undefined;
+            }
+        },
+        remove(key) {
+            ensureLocalStorageIsAvailable();
+            localStorage.removeItem(`${prefix}:${key}`);
+        },
+        clear() {
+            ensureLocalStorageIsAvailable();
+            Object.keys(localStorage).forEach((key) => {
+                if (key.startsWith(prefix)) {
+                    localStorage.removeItem(key);
+                }
+            });
+        },
+    };
+
+    function ensureLocalStorageIsAvailable(): void {
```

Comment:
I know why you put the return statement before the function definition (public stuff first, implementation details later), but this could easily confuse people not familiar with JS's quirks.

Therefore, I recommend placing the `ensureLocalStorageIsAvailable` before the return statement, or even moving it outside of `createLocalStorageDataStore` altogether (I also realize that you put it inside to localize all the `localStorage`-specific stuff into a single function, but it's not a big deal to move it outside IMO).